src/bin-utils/index.test.js

```diff
@@ -305,6 +311,69 @@ test(
   },
 )
 
+test('specificHelpScript: returns no script available', () => {
+  const config = {scripts: {}}
+  const message = specificHelpScript(config, 'foo')
+  const expected = chalk.yellow('Script foo does not exist')
+  expect(message).toBe(expected)
+})
+
+test(
+  'specificHelpScript: do not display script with flag hiddenFromHelp=true',
+  () => {
+    const config = {
+      scripts: {
+        foo: {
+          description: 'the foo script',
+          script: 'echo "foo"',
+          hiddenFromHelp: true,
+        },
+      },
+    }
+    const message = specificHelpScript(config, 'foo')
+    const expected = chalk.yellow('Script foo does not exist')
+    expect(message).toBe(expected)
+  },
+)
+
+test('specificHelpScript: formats a nice message', () => {
+  const config = {
```

Comment:
Thanks for the nice explanation for prefix-matches. Makes sense. 
Let us assume this as config.
```
{
   scripts : {
       watch: {
		js: 'watch javascript',
		css: 'watch css'
	},
	build: 'build things'
   }
}
```
So from what I understand, in this case, `nps help w.j` and `nps help w.c` should return help definitions. But `nps help w` should not. (Even if prefix-matches has returned a non-empty array). Using prefix-matches will not be enough, IMO. I will also need to check if w.j is deterministic enough to resolve as a script. Am I missing something here?

Regarding script being a number, actually, I just wanted to gracefully handle invalid data (in case). There were some test cases with `script : 42`.